src/platform/site-wide/announcements/components/ScheduledMaintenance.jsx

```diff
@@ -0,0 +1,102 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment-timezone';
+import PromoBanner, {
+  PROMO_BANNER_TYPES,
+} from '@department-of-veterans-affairs/formation-react/PromoBanner';
+
+class ScheduledMaintenance extends Component {
+  static propTypes = {
+    announcement: PropTypes.shape({
+      downtimeStartsAt: PropTypes.object.isRequired,
+      expiresAt: PropTypes.object.isRequired,
+    }).isRequired,
+    dismiss: PropTypes.func.isRequired,
+  };
+
+  componentWillUnmount() {
+    // Prevent memory leaks by clearing timeouts.
+    clearTimeout(this.rerenderTimeout);
+  }
+
+  refreshIn = milliseconds => {
+    clearTimeout(this.rerenderTimeout);
+    this.rerenderTimeout = setTimeout(() => {
+      this.forceUpdate();
+    }, milliseconds);
+  };
```

Comment:
No need to refactor, but I think an easier way to manage timeouts is to set an interval during `componentDidMount` then set the banner text during each tick using state.